llm_perf/hardware/update_llm_perf_intel.py

```diff
@@ -0,0 +1,219 @@
+
+SUBSET = os.getenv("SUBSET", None)
+MACHINE = os.getenv("MACHINE", None)
+BACKEND = os.getenv("BACKEND", None)
+HARDWARE = "intel"
+
+if os.getenv("MACHINE", None) is None and os.getenv("SUBSET", None) is None:
+    PUSH_REPO_ID = f"optimum-benchmark/llm-perf-{BACKEND}-intel-debug"
+    CANONICAL_PRETRAINED_OPEN_LLM_LIST = ["gpt2"]
+    SUBSET = "unquantized"
+elif os.getenv("MACHINE", None) is not None and os.getenv("SUBSET", None) is not None:
+    PUSH_REPO_ID = f"optimum-benchmark/llm-perf-{BACKEND}-intel-{SUBSET}-{MACHINE}"
+else:
+    raise ValueError("Either both MACHINE and SUBSET should be set for benchmarking or neither for debugging")
+
+ATTENTION_CONFIGS = ["eager", "sdpa"]
+
+
+if SUBSET == "unquantized":
+    WEIGHTS_CONFIGS = {
+        # unquantized
+        "float32": {"torch_dtype": "float32", "quant_scheme": None, "quant_config": {}},
+        "float16": {"torch_dtype": "float16", "quant_scheme": None, "quant_config": {}},
+        "bfloat16": {"torch_dtype": "bfloat16", "quant_scheme": None, "quant_config": {}},
+    }
+elif SUBSET == "bnb":
+    WEIGHTS_CONFIGS = {
+        # bnb
+        "4bit-bnb": {"torch_dtype": "float16", "quant_scheme": "bnb", "quant_config": {"load_in_4bit": True}},
+        "8bit-bnb": {"torch_dtype": "float16", "quant_scheme": "bnb", "quant_config": {"load_in_8bit": True}},
+    }
+elif SUBSET == "gptq":
+    WEIGHTS_CONFIGS = {
+        # gptq
+        "4bit-gptq-exllama-v1": {
+            "quant_scheme": "gptq",
+            "torch_dtype": "float16",
+            "quant_config": {"bits": 4, "use_exllama ": True, "version": 1, "model_seqlen": 256},
+        },
+        "4bit-gptq-exllama-v2": {
+            "torch_dtype": "float16",
+            "quant_scheme": "gptq",
+            "quant_config": {"bits": 4, "use_exllama ": True, "version": 2, "model_seqlen": 256},
+        },
+    }
+elif SUBSET == "awq":
+    WEIGHTS_CONFIGS = {
+        # awq
+        "4bit-awq-gemm": {
+            "torch_dtype": "float16",
+            "quant_scheme": "awq",
+            "quant_config": {"bits": 4, "version": "gemm"},
+        },
+        "4bit-awq-gemv": {
+            "torch_dtype": "float16",
+            "quant_scheme": "awq",
+            "quant_config": {"bits": 4, "version": "gemv"},
+        },
+        "4bit-awq-exllama-v1": {
+            "torch_dtype": "float16",
+            "quant_scheme": "awq",
+            "quant_config": {
+                "bits": 4,
+                "version": "exllama",
+                "exllama_config": {"version": 1, "max_input_len": 64, "max_batch_size": 1},
+            },
+        },
+        "4bit-awq-exllama-v2": {
+            "torch_dtype": "float16",
+            "quant_scheme": "awq",
+            "quant_config": {
+                "bits": 4,
+                "version": "exllama",
+                "exllama_config": {"version": 2, "max_input_len": 64, "max_batch_size": 1},
+            },
+        },
+    }
```

Comment:
most of these configs only work with pytorch+cuda transformers.

Reply:
Thanks indeed i forgot to delete them